### ISSUE TITLE
Hashtag shortcut from external instances include muted accounts

### ISSUE THREAD

**Issue by eliocamp** [cheeaun/phanpy#371]
**Describe the bug**

As described in the title. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Add a shortcut for a hashtag from your instance.
2. Add a shortcut for the same hashtag but from an external instance. 
3. See a post that is in both columns and mute the user. 
4. Reload both columns. The post disappears on the first column but is there on the second one. 

**Expected behavior**

Muted accounts should not be shown. 

**Screenshots**

Shortcuts for the #argentina hashtag. One on mastodon.social (my instance) and the other on mas.to. Both show posts by the AllQuakes bot

![2023-12-20_10-41](https://github.com/cheeaun/phanpy/assets/8617595/a31a96cb-609b-47d4-8e5d-a55b7c276e5f)

Muting the bot

![2023-12-20_10-42](https://github.com/cheeaun/phanpy/assets/8617595/d6e2080f-28b1-4e45-adad-eefd6dbc27f4)

Now the posts are not there on the mastodon.social column but is still shown in the mas.to column. 

![2023-12-20_10-42_1](https://github.com/cheeaun/phanpy/assets/8617595/96b864f5-0f50-44a8-a41e-4e06499a394d)

**Comment by cheeaun** [comment]
Muted accounts are stored in the back-end of your logged-in instance, not really a client-side setting. It doesn't work on the other instance.

**Comment by eliocamp** [comment]
Ah, I was afraid that this might be not technically possible. Could phanpy retrieve and/or store the list of muted accounts and apply the filter itself? It might be too cumbersome and hard to maintain.

**Comment by cheeaun** [comment]
Storing the list on client-side is… yeah, cumbersome 🙈

Some mobile apps are probably already doing it (e.g. Ivory with client-side filters, mutes, etc — I'm not sure if they also apply it on remote timelines) with offline⇔remote database sync. Phanpy doesn't have this kind of db + sync implementation yet. Mutes list can also be _very_ long with paginated responses (I'm not sure what's the limit or if there's even a limit?)

**Comment by cheeaun** [comment]
@eliocamp if the user has an account on that instance and user has added the account on Phanpy, the mute list (from that instance) should work theoretically. Though the user will need to mute the same profile from both accounts.

Note that the posts are not hidden from the client-side by checking the mute list. It's the back-end/server that's doing the work, by not including the muted posts from the beginning.

**Comment by eliocamp** [comment]
> @eliocamp if the user has an account on that instance and user has added the account on Phanpy, the mute list (from that instance) should work theoretically. Though the user will need to mute the same profile from both accounts.

That's my case. I have an account on mastodon.social and one on sigmoid.social. If I add a shortcut on phanpy in my sogmoid.social that points to the mastodon.social instance, it doesn't use the mastodon.social mute list.

**Comment by cheeaun** [comment]
> If I add a shortcut on phanpy in my sogmoid.social that points to the mastodon.social instance, it doesn't use the mastodon.social mute list.

Yeah, you'll have to mute the same accounts on the other instance. Or… you could export/import the mute list between the instances… 🤔

**Comment by eliocamp** [comment]
Or not. If I add the hashtag shortcut using that performs the request from another account, I can use that account mute list only and I don't need to keep them in sync.
